subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps_utils.c

```diff
@@ -99,6 +99,24 @@ static int settings_set(const char *key, size_t len_rd,
 	return -ENOTSUP;
 }
 
+int npgps_lock(void)
+{
+	int err = k_sem_take(&pgps_active, K_NO_WAIT);
+
+	if (!err) {
+		LOG_DBG("pgps_active LOCKED");
+	} else {
+		LOG_ERR("Unable to lock pgps_active: %d", err);
+	}
+	return err;
+}
+
+void npgps_unlock(void)
+{
+	k_sem_give(&pgps_active);
+	LOG_DBG("pgps_active UNLOCKED");
+}
+
```

Comment:
Is this semaphore just to prevent multiple threads from attempting to perform a PGPS download simultaneously? A comment explaining what it protects would be useful I think